CorsixTH/Lua/dialogs/resizables/menu_list_dialogs/custom_campaign.lua

```diff
@@ -33,42 +33,63 @@ local col_scrollbar = {
 local details_width = 280
 
 --! Collect the campaign levels at the provided path
---!param path (str) File system path to search.
---!return (array) The found levels, with some basic information about each level.
-local function createCampaignList(path)
-  local campaigns = {}
-
-  for file in lfs.dir(path) do
-    if file:match("%.campaign$") then
-      local campaign_info, err = TheApp:readCampaignFile(file)
-      if not campaign_info then
-        print(err)
+--!param paths_table (array) File system paths to search.
+--!return (array) The found levels, with some basic information about each level, in alphabetical order.
```

Comment:
It also returns a duplicates flag

Technically the description at line 35 isn't complete, it also checks for validity of the campaign and reports errors to the log